modules/command/dh/dh-process-blockchain-events-command.js

```diff
@@ -0,0 +1,148 @@
+const Command = require('../command');
+const Utilities = require('../../Utilities');
+const Models = require('../../../models/index');
+const constants = require('../../constants');
+
+/**
+ * Repeatable command that checks whether offer is created or litigation is successfully initiated
+ */
+class DHProcessBlockchainEventsCommand extends Command {
+    constructor(ctx) {
+        super(ctx);
+        this.config = ctx.config;
+        this.logger = ctx.logger;
+        this.commandExecutor = ctx.commandExecutor;
+        this.blockchain = ctx.blockchain;
+        this.profileService = ctx.profileService;
+        this.errorNotificationService = ctx.errorNotificationService;
+        this.dhService = ctx.dhService;
+    }
+
+    /**
+     * Executes command and produces one or more events
+     * @param command
+     */
+    async execute(command) {
+        try {
+            const events = await Models.events.findAll({
+                where: {
+                    event: {
+                        [Models.Sequelize.Op.in]: [
+                            constants.EVENT_NAME.LitigationInitiated,
+                            constants.EVENT_NAME.OfferCreated],
+                    },
+                    finished: 0,
+                },
+            });
+            if (events) {
+                const litigationInitiatedEvents = events
+                    .filter(e => e.event === constants.EVENT_NAME.LitigationInitiated);
+
+                const offerCreatedEvents = events
+                    .filter(e => e.event === constants.EVENT_NAME.OfferCreated);
+
+                await this.handleLitigationInitiatedEvents(litigationInitiatedEvents);
+
+                await this.handleOfferCreatedEvents(offerCreatedEvents);
+            }
+        } catch (e) {
+            this.logger.error(`Failed to process dhProcessBlockchainEventsCommand. ${e}`);
+            this.errorNotificationService.notifyError(
+                e,
+                null,
+                constants.PROCESS_NAME.bcEventsHandling,
+            );
+        }
+
+        return Command.repeat();
+    }
+
+    async handleLitigationInitiatedEvents(events) {
+        const allMyIdentities = {};
+        this.blockchain.getAllBlockchainIds()
+            .forEach(id => allMyIdentities[id] = this.profileService.getIdentity(id));
+        const event = events.find((e) => {
```

Comment:
What if there are several events for an identity? find vs filter

Reply:
Next event will be handled in next iteration of command, we are handling litigation events one by one.